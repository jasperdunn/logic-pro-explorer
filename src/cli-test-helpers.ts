import { execSync, spawnSync } from 'child_process'
import { dirname, resolve } from 'path'
import { fileURLToPath } from 'url'

export const packagePath = resolve(dirname(fileURLToPath(import.meta.url)), '../dist/index.js')

export function runCommand(command: string): string {
  return execSync(`node ${packagePath} ${command}`).toString()
}

export function runSpawnedCommand(command: string, input = ''): string {
  const result = spawnSync('node', [packagePath, command], { input })
  if (result.status !== 0) {
    throw new Error(`Command '${command}' failed with exit code ${result.status}.`)
  }
  return result.stdout.toString()
}
