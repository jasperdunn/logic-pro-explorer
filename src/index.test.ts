import { runCommand } from './cli-test-helpers.js'

beforeAll(() => {
  runCommand('config -p /Volumes/data-ssd/audio/Logic/')
})

describe('help command', () => {
  it('should display the help message', () => {
    expect(runCommand('help')).toMatchSnapshot()
  })
})
