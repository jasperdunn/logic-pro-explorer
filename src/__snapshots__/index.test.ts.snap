// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`help command > should display the help message 1`] = `
"Usage: logic-pro-explorer [command] [options]

A CLI tool to query and analyze .logicx project files on your system.

Options:
  -v, --version     Displays the version number.
  -h, --help        Displays this help message.

Commands:
  config [options]  Set configuration options.
  tree [options]    Displays a tree of Logic projects or components.
  find              Find projects that are using the selected components.
  info [options]    Display information about the selected project or
                    component.
  uninstall         Remove logic-pro-explorer from your computer.
  help              Displays this help message.
"
`;
